### PR TITLE
[MOB-1127] Set up Github Actions CI

### PR DESCRIPTION
**Pull request by vbabenkoru**
Set up testing and building on CI as a first step towards easier deploys.

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        java-version: 1.8
+    - name: Grant execute permission for gradlew
+      run: chmod +x gradlew
+    - name: Test with Gradle
+      run: ./gradlew test
+    - name: Build with Gradle
+      run: ./gradlew build
+    - name: Deploy to AWS
+      uses: appleboy/lambda-action@master
+      with:
```
Comment (ktham) [inline .github/workflows/gradle.yml]: how about we replace the `with: ` part with this:

```
        env:
          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
        run: aws lambda update-function-code --function-name mparticle_iterable_firehose_test --zip-file fileb://iterable-extension/build/distributions/iterable-extension.zip
```

This way, we just use the `awscli` that comes with the Github action container and avoid using external git repos

Comment (ktham) [reply .github/workflows/gradle.yml]: The `https://github.com/stcalica/update-lambda/blob/master/index.js` that you suggested does the same thing in JS 

```
const core = require('@actions/core');
const github = require('@actions/github');
const fs = require('fs');
const AWS = require('aws-sdk');

try {
  const functionName = core.getInput('function-name');
  const package = core.getInput('package');
  const AWS_SECRET_KEY = core.getInput('AWS_SECRET_KEY');
  const AWS_SECRET_ID = core.getInput('AWS_SECRET_ID');
  const AWS_REGION = core.getInput('AWS_REGION');

  console.log(`Deploying ${functionName} from ${package}.`);

  var zipBuffer = fs.readFileSync(`./${package}`);
  core.debug('ZIP file put into memory buffer.');

  const lambda = new AWS.Lambda({
      apiVersion: '2015-03-31',
      region: AWS_REGION,
      secretAccessKey: AWS_SECRET_KEY,
      accessKeyId: AWS_SECRET_ID,
      maxRetries: 3,
      sslEnabled: true,
      logger: console,
  });

  const params = {
    FunctionName: functionName,
    Publish: true,
    ZipFile: zipBuffer,
  };

  lambda.updateFunctionCode(params, err => {
      if (err) {
          console.error(err);
          core.setFailed(err)
      }
  });

} catch (error) {
  core.setFailed(error.message);
}
```

Essentially, we are just calling `update-function-code` in either case, which can be done with a single `aws-cli` call

Comment (ktham) [reply .github/workflows/gradle.yml]: We can also create a `Deploy to prd` step with the conditional `if: github.ref == 'master'` if you'd like

Comment (vbabenkoru) [reply .github/workflows/gradle.yml]: Updated, it works!

**Comment by vbabenkoru** [comment]
Modified to only deploy on pushes to `develop`.

**Pull request merged**